### PR TITLE
[rv_dm,sival] RV_DM post silicon validation testplan

### PR DESCRIPTION
**Pull request by GregAC**
This adds features to the rv_dm data, moves the tests to a separate file and expands them for post silicon validation.

### CODE REVIEW

**Comment by pamaury** [comment]
Should there be a feature for NDM (non-debug reset) specifically?

File: hw/top_earlgrey/data/ip/chip_rv_dm_testplan.hjson [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+            - Shuffle the list of CSRs yet again.
+            - Read all CSRs back and check their values for correctness while adhering to the CSR's
+              access policies.
+            - Accesses to CSRs external to `rv_dm` go through RV_DM SBA interface into the `xbar`.
+            '''
+      stage: V2
+      si_stage: SV2
+      tests: ["chip_jtag_csr_rw"]
+      lc_states: ["TEST_UNLOCKED", "DEV", "RMA"]
+      host_support: "true"
```
Comment (pamaury) [inline hw/top_earlgrey/data/ip/chip_rv_dm_testplan.hjson:24]: I have not seen `host_support` before, what does it mean exactly? Is there a place where the valid attribute are defined?

File: hw/ip/rv_dm/data/rv_dm.hjson [hunk 2]
```diff
@@ -119,6 +119,102 @@
...
+                    available hardware breakpoints
+                  - Single stepping
+                  - Reading and writing all registers
+                  - Reading and writing CSRs
+                  - Reading and writing memory
+                  - Setting the PC to jump to some location
+                  - Executing code from GDB (using the call command)
+             '''
+    }
+  ]
```
Comment (pamaury) [inline hw/ip/rv_dm/data/rv_dm.hjson:217]: Should non-debug reset (NDM) be a feature?

File: hw/top_earlgrey/data/ip/chip_rv_dm_testplan.hjson (hunk 1 above)
Comment (GregAC) [reply hw/top_earlgrey/data/ip/chip_rv_dm_testplan.hjson:24]: It means the test will require a component that's outside of the device itself to run. It's defined in this document: https://docs.google.com/document/d/14b1RWc2jozg9CfuFK4xCUMR5j4gCjrgYrif40Irqc7Q/edit#heading=h.4cj1geh4tmef

File: hw/ip/rv_dm/data/rv_dm.hjson (hunk 2 above)
Comment (GregAC) [reply hw/ip/rv_dm/data/rv_dm.hjson:217]: Good point, yes I think it should

**Review by pamaury — APPROVED** [review]
Looks good to me, apart from the changes to quick lint that you probably did not want to commit?

**Comment by GregAC** [comment]
> Looks good to me, apart from the changes to quick lint that you probably did not want to commit?

Whoops! Well spotted

File: hw/top_earlgrey/data/ip/chip_rv_dm_testplan.hjson [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+        "RV_DM.JTAG.FSM",
+        "RV_DM.JTAG.DTM"
+      ]
+    }
+    {
+      name: chip_sw_rv_dm_access_after_hw_reset
+      desc: '''Verify RV_DM works after a watchdog or escalated reset.
+
+            - Access some RV_DM CSRs both before and after resets.
+            - An activation would be required, and the tap strap would also be sampled again.
```
Comment (pamaury) [inline hw/top_earlgrey/data/ip/chip_rv_dm_testplan.hjson:162]: What exactly does this sentence mean?

File: hw/top_earlgrey/data/ip/chip_rv_dm_testplan.hjson [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+      lc_states: ["TEST_UNLOCKED", "DEV", "RMA"]
+      host_support: "true"
+      features: [
+        "RV_DM.JTAG.FSM",
+        "RV_DM.JTAG.DTM"
+      ]
+    }
+    {
+      name: chip_sw_rv_dm_access_after_hw_reset
+      desc: '''Verify RV_DM works after a watchdog or escalated reset.
```
Comment (pamaury) [inline hw/top_earlgrey/data/ip/chip_rv_dm_testplan.hjson:159]: For SiVal, is there a way to simulate an escalated reset?

File: hw/top_earlgrey/data/ip/chip_rv_dm_testplan.hjson [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+    }
+    {
+      name: chip_rv_dm_perform_debug
+      desc: '''
+            - X-ref'ed with rom_e2e_jtag_inject from rom testplan.
+            - X-ref'ed with chip_sw_flash_lc_iso_part_sw_wr_en.
+            - X-ref'ed with manuf_cp_device_info_flash_wr from manufacturing testplan.
+            - Using the sram injection mechanism from rom_e2e_jtag_inject, load a SRAM program that
+              writes to isolated flash partition while the device is in TEST_UNLOCKED state.
+            - After writing, verify that the test program cannot read back the written value.
```
Comment (pamaury) [inline hw/top_earlgrey/data/ip/chip_rv_dm_testplan.hjson:69]: It is not clear to me what this means. The SRAM program writes something to the flash. Then the `test program` (is this another program that is bootstraped?) needs to try and read it?

File: hw/top_earlgrey/data/ip/chip_rv_dm_testplan.hjson [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+{
+  name: rv_dm
+  testpoints: [
+    // RV_DM (JTAG) tests:
+    {
+      name: chip_jtag_csr_rw
+      desc: '''
+            Verify accessibility of all the CSRs in the chip over JTAG.
+
+            - Shuffle the list of CSRs first to remove the effect of ordering.
```
Comment (pamaury) [inline hw/top_earlgrey/data/ip/chip_rv_dm_testplan.hjson:13]: Is that literally *all* of the CSRs? How do we handle CSRs that have nontrivial side-effects?

**Comment by GregAC** [comment]
@pamaury are you happy to merge this as is or do you want to conclude the discussions around your questions?

**Comment by pamaury** [comment]
Yes you can merge

**Comment by engdoreis** [comment]
CHANGE AUTHORIZED: hw/ip/rv_dm/data/rv_dm.hjson

**Comment by marnovandermaas** [comment]
CHANGE AUTHORIZED: hw/ip/rv_dm/data/rv_dm.hjson

**Pull request merged**